kpatch-build/lookup.c

```diff
@@ -108,9 +107,15 @@ static int locals_match(struct lookup_table *table, int idx,
 			continue;
 
 		found = 0;
-		for (child = child_locals; child->name; child++) {
-			if (child->type == sym->type &&
-			    !strcmp(child->name, sym->name)) {
+		iter_sym = file_sym;
```

Comment:
The variable names were already fairly confusing in this function, but the addition of `struct symbol` to this file confuses the naming even more.  There's no logical difference between `sym` and `iter_sym` when reading the code.

I'd suggest making the following naming changes here, as well as in other places with `struct object_symbol *sym`.

- rename `sym` to `table_sym`.
- rename `iter_sym` to `sym`.

That way, `sym` has its traditional meaning (an instance of `struct symbol *`) and `table_sym` is more appropriately named.

Reply:
Thanks, good suggestion. I was struggling with variable naming in this patchset, at some point I had lookup_sym->lookup_sym in lookup_local_symbol().